I wrote the unit tests in #7, and it seems that it works except where decimals are involved. For example (in my branch):
```
FAIL: test_timeparse_31 (tests.TestRelativeDeltaOutput)
timeparse test case 31.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 740, in test_timeparse_31
    self.assertEqual(timeparse.parse('5.6 wk', as_timedelta=True), relativedelta(seconds = 3386880))
AssertionError: relativedelta(days=+39) != relativedelta(days=+39, hours=+4, minutes=+48)

======================================================================
FAIL: test_timeparse_9 (tests.TestRelativeDeltaOutput)
timeparse test case 9.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 531, in test_timeparse_9
    self.assertEqual(timeparse.parse('4:13:02.266', as_timedelta=True), relativedelta(seconds = 15182, microseconds=266 * 1000))
AssertionError: relativedelta(hours=+4, minutes=+13, seconds=+2) != relativedelta(hours=+4, minutes=+13, seconds=+2, microseconds=+266000)
```

<details>
<summary>All in all, I end up with 21 failed cases with either this or what appears to be similar issues:</summary>

```
> python .\tests.py
........................................................F.F.F.F....FFFFFFFF.FF.....F....FFF.....F...F...F
======================================================================
FAIL: test_combined (tests.TestRelativeDeltaOutput)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 769, in test_combined
    self.assertEqual(timeparse.parse('1y2mo3w4d5h6m7s8ms', as_timedelta=True), relativedelta(seconds = 38898367, microseconds=8000))
AssertionError: relativedelta(years=+1, months=+2, days=+25, hours=+5, [38 chars]8000) != relativedelta(days=+450, hours=+5, minutes=+6, seconds=[18 chars]8000)

======================================================================
FAIL: test_plain_numbers (tests.TestRelativeDeltaOutput)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 764, in test_plain_numbers
    self.assertEqual(timeparse.parse('10.1', as_timedelta=True), relativedelta(seconds = 10, microsecond = 100000))
AssertionError: relativedelta(seconds=+10) != relativedelta(seconds=+10, microsecond=100000)

======================================================================
FAIL: test_timeparse_10 (tests.TestRelativeDeltaOutput)
timeparse test case 10.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 537, in test_timeparse_10
    self.assertEqual(timeparse.parse('2:04:13:02.266', as_timedelta=True),
AssertionError: relativedelta(days=+2, hours=+4, minutes=+13, seconds=+2) != relativedelta(days=+2, hours=+4, minutes=+13, seconds=+2, microsecond=266000)

======================================================================
FAIL: test_timeparse_12 (tests.TestRelativeDeltaOutput)
timeparse test case 12.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 584, in test_timeparse_12
    self.assertEqual(timeparse.parse('2 days,  4:13:02.266', as_timedelta=True),
AssertionError: relativedelta(days=+2, hours=+4, minutes=+13, seconds=+2) != relativedelta(days=+2, hours=+4, minutes=+13, seconds=+2, microsecond=266000)

======================================================================
FAIL: test_timeparse_16b (tests.TestRelativeDeltaOutput)
timeparse test case 16b.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 626, in test_timeparse_16b
    self.assertEqual(timeparse.parse('1.75 s', as_timedelta=True), relativedelta(seconds = 1, microsecond = 10000 * 75))
AssertionError: relativedelta(seconds=+1) != relativedelta(seconds=+1, microsecond=750000)

======================================================================
FAIL: test_timeparse_16c (tests.TestRelativeDeltaOutput)
timeparse test case 16c.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 632, in test_timeparse_16c
    self.assertEqual(timeparse.parse('1.75 sec', as_timedelta=True), relativedelta(seconds = 1, microsecond = 10000 * 75))
AssertionError: relativedelta(seconds=+1) != relativedelta(seconds=+1, microsecond=750000)

======================================================================
FAIL: test_timeparse_16d (tests.TestRelativeDeltaOutput)
timeparse test case 16d.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 638, in test_timeparse_16d
    self.assertEqual(timeparse.parse('1.75 secs', as_timedelta=True), relativedelta(seconds = 1, microsecond = 75 * 10000))
AssertionError: relativedelta(seconds=+1) != relativedelta(seconds=+1, microsecond=750000)

======================================================================
FAIL: test_timeparse_16e (tests.TestRelativeDeltaOutput)
timeparse test case 16e.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 644, in test_timeparse_16e
    self.assertEqual(timeparse.parse('1.75 second', as_timedelta=True), relativedelta(seconds = 1, microsecond = 75 * 10000))
AssertionError: relativedelta(seconds=+1) != relativedelta(seconds=+1, microsecond=750000)

======================================================================
FAIL: test_timeparse_16f (tests.TestRelativeDeltaOutput)
timeparse test case 16f.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 650, in test_timeparse_16f
    self.assertEqual(timeparse.parse('1.75 seconds', as_timedelta=True), relativedelta(seconds = 1, microsecond = 75 * 10000))
AssertionError: relativedelta(seconds=+1) != relativedelta(seconds=+1, microsecond=750000)

======================================================================
FAIL: test_timeparse_17 (tests.TestRelativeDeltaOutput)
timeparse test case 17.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 656, in test_timeparse_17
    self.assertEqual(timeparse.parse('1.2 m', as_timedelta=True), relativedelta(seconds = 72))
AssertionError: relativedelta(minutes=+1) != relativedelta(minutes=+1, seconds=+12)

======================================================================
FAIL: test_timeparse_18 (tests.TestRelativeDeltaOutput)
timeparse test case 18.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 662, in test_timeparse_18
    self.assertEqual(timeparse.parse('1.2 min', as_timedelta=True), relativedelta(seconds = 72))
AssertionError: relativedelta(minutes=+1) != relativedelta(minutes=+1, seconds=+12)

======================================================================
FAIL: test_timeparse_19 (tests.TestRelativeDeltaOutput)
timeparse test case 19.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 668, in test_timeparse_19
    self.assertEqual(timeparse.parse('1.2 mins', as_timedelta=True), relativedelta(seconds = 72))
AssertionError: relativedelta(minutes=+1) != relativedelta(minutes=+1, seconds=+12)

======================================================================
FAIL: test_timeparse_20 (tests.TestRelativeDeltaOutput)
timeparse test case 20.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 674, in test_timeparse_20
    self.assertEqual(timeparse.parse('1.2 minute', as_timedelta=True), relativedelta(seconds = 72))
AssertionError: relativedelta(minutes=+1) != relativedelta(minutes=+1, seconds=+12)

======================================================================
FAIL: test_timeparse_21 (tests.TestRelativeDeltaOutput)
timeparse test case 21.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 680, in test_timeparse_21
    self.assertEqual(timeparse.parse('1.2 minutes', as_timedelta=True), relativedelta(seconds = 72))
AssertionError: relativedelta(minutes=+1) != relativedelta(minutes=+1, seconds=+12)

======================================================================
FAIL: test_timeparse_27 (tests.TestRelativeDeltaOutput)
timeparse test case 27.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 716, in test_timeparse_27
    self.assertEqual(timeparse.parse('1.24 days', as_timedelta=True), relativedelta(seconds = 107136))
AssertionError: relativedelta(days=+1) != relativedelta(days=+1, hours=+5, minutes=+45, seconds=+36)

======================================================================
FAIL: test_timeparse_31 (tests.TestRelativeDeltaOutput)
timeparse test case 31.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 740, in test_timeparse_31
    self.assertEqual(timeparse.parse('5.6 wk', as_timedelta=True), relativedelta(seconds = 3386880))
AssertionError: relativedelta(days=+39) != relativedelta(days=+39, hours=+4, minutes=+48)

======================================================================
FAIL: test_timeparse_32 (tests.TestRelativeDeltaOutput)
timeparse test case 32.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 746, in test_timeparse_32
    self.assertEqual(timeparse.parse('5.6 week', as_timedelta=True), relativedelta(seconds = 3386880))
AssertionError: relativedelta(days=+39) != relativedelta(days=+39, hours=+4, minutes=+48)

======================================================================
FAIL: test_timeparse_33 (tests.TestRelativeDeltaOutput)
timeparse test case 33.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 752, in test_timeparse_33
    self.assertEqual(timeparse.parse('5.6 weeks', as_timedelta=True), relativedelta(seconds = 3386880))
AssertionError: relativedelta(days=+39) != relativedelta(days=+39, hours=+4, minutes=+48)

======================================================================
FAIL: test_timeparse_9 (tests.TestRelativeDeltaOutput)
timeparse test case 9.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 531, in test_timeparse_9
    self.assertEqual(timeparse.parse('4:13:02.266', as_timedelta=True), relativedelta(seconds = 15182, microseconds=266 * 1000))
AssertionError: relativedelta(hours=+4, minutes=+13, seconds=+2) != relativedelta(hours=+4, minutes=+13, seconds=+2, microseconds=+266000)

======================================================================
FAIL: test_timeparse_granularity_3 (tests.TestRelativeDeltaOutput)
Check that minute-level granularity does not apply inappropriately.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 558, in test_timeparse_granularity_3
    self.assertEqual(timeparse.parse('7:02.223', granularity='minutes', as_timedelta=True), relativedelta(seconds = 2, microsecond = 1000 * 223, minute=7))
AssertionError: relativedelta(minutes=+7, seconds=+2) != relativedelta(seconds=+2, minute=7, microsecond=223000)

======================================================================
FAIL: test_timeparse_unparsed (tests.TestRelativeDeltaOutput)
Check that unparsed values tries to converts into int().
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\abold\Desktop\pytimeparse2\tests.py", line 571, in test_timeparse_unparsed
    self.assertEqual(timeparse.parse(-18.333, as_timedelta=True), relativedelta(seconds = -18, microsecond = -333 * 1000))
AssertionError: relativedelta(seconds=-18) != relativedelta(seconds=-18, microsecond=-333000)

----------------------------------------------------------------------
Ran 105 tests in 0.043s

FAILED (failures=21)
```
</details>

It seems that somehow, the decimals are throwing the code for a loop. I may look into that a bit more after finals... (though that's what I said about writing this... 😔)

--- a/pytimeparse2.py
+++ b/pytimeparse2.py
@@ -109,7 +109,7 @@ def _all_digits(mdict, delta_class):
     if HAS_RELITIVE_TIMEDELTA and issubclass(delta_class, relativedelta):
         if 'milliseconds' in mdict:
             mdict['microseconds'] = float(mdict.pop('milliseconds') or 0) * 1000
-        return delta_class(**{k: float(v) for k, v in mdict.items() if v})
+        return delta_class(**{k: float(v) for k, v in mdict.items() if v}).normalized()
 
     delta = delta_class(**{
         key: float(mdict.pop(key) or 0)
@@ -257,7 +257,7 @@ def parse(
                 return int(new_value)
             else:
                 return new_value
-        return value.normalized()
+        return value
     except Exception:
         if raise_exception:
             raise

--- a/pytimeparse2.py
+++ b/pytimeparse2.py
@@ -257,7 +257,7 @@ def parse(
                 return int(new_value)
             else:
                 return new_value
-        return value
+        return value.normalized()
     except Exception:
         if raise_exception:
             raise

--- a/tests.py
+++ b/tests.py
@@ -14,16 +14,13 @@ import doctest
 import re
 import pytimeparse2 as timeparse
 import unittest
+from dateutil.relativedelta import relativedelta
 
 
-class TestTimeparse(unittest.TestCase):
+class TestParsing(unittest.TestCase):
     """
-    Unit tests for the `timeparse` module.
+    Unit tests for basic regex mat
     """
-
-    def setUp(self):
-        """Setup function."""
-        pass
 
     def test_mins(self):
         """Test parsing minutes."""
@@ -109,6 +106,13 @@ class TestTimeparse(unittest.TestCase):
             set(re.match(timeparse.TIMEFORMATS[0] + r'\s*$',
                          '16h32m64s  ').groupdict().items()),
             set([('hours', '16'), ('minutes', '32'), ('seconds', '64')]))
+
+
+
+class TestNumberOutput(unittest.TestCase):
+    """
+    Unit tests to ensure that numerical outputs are correct
+    """
 
     def test_timeparse_multipliers(self):
         """Test parsing time unit multipliers."""
@@ -428,6 +432,319 @@ class TestTimeparse(unittest.TestCase):
     def test_combined(self):
         self.assertEqual(timeparse.parse('1y2mo3w4d5h6m7s8ms'), 38898367.008)
 
+
+
+
+class TestRelativeDeltaOutput(unittest.TestCase):
+    """
+    Unit tests to ensure that numerical outputs are correct
+    """
+
+    def test_timeparse_multipliers(self):
+        """Test parsing time unit multipliers."""
+        self.assertEqual(timeparse.parse('32 min', as_timedelta=True), relativedelta(minutes=32))
+        self.assertEqual(timeparse.parse('1 min', as_timedelta=True), relativedelta(minutes=1))
+        self.assertEqual(timeparse.parse('1 hours', as_timedelta=True), relativedelta(hours=1))
+        self.assertEqual(timeparse.parse('1 day', as_timedelta=True), relativedelta(days=1))
+        self.assertEqual(timeparse.parse('1 sec', as_timedelta=True), relativedelta(seconds=1))
+
+    def test_timeparse_signs(self):
+        """Test parsing time signs."""
+        self.assertEqual(timeparse.parse('+32 m 1 s', as_timedelta=True), relativedelta(minutes=32, seconds=1))
+        self.assertEqual(timeparse.parse('+ 32 m 1 s', as_timedelta=True), relativedelta(minutes=32, seconds=1))
+        self.assertEqual(timeparse.parse('-32 m 1 s', as_timedelta=True), -relativedelta(minutes=32, seconds=1))
+        self.assertEqual(timeparse.parse('- 32 m 1 s', as_timedelta=True), -relativedelta(minutes=32, seconds=1))
+        self.assertIsNone(timeparse.parse('32 m - 1 s', as_timedelta=True))
+        self.assertIsNone(timeparse.parse('32 m + 1 s', as_timedelta=True))
+
+    def test_timeparse_1(self):
+        """timeparse test case 1."""
+        self.assertEqual(timeparse.parse('32m', as_timedelta=True), relativedelta(minutes=32))
+        self.assertEqual(timeparse.parse('+32m', as_timedelta=True), relativedelta(minutes=32))
+        self.assertEqual(timeparse.parse('-32m', as_timedelta=True), -relativedelta(minutes=32))
+
+    def test_timeparse_2(self):
+        """timeparse test case 2."""
+        self.assertEqual(timeparse.parse('2h32m', as_timedelta=True), relativedelta(hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('+2h32m', as_timedelta=True), relativedelta(hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('-2h32m', as_timedelta=True), -relativedelta(hours=2, minutes=32))
+
+    def test_timeparse_3(self):
+        """timeparse test case 3."""
+        self.assertEqual(timeparse.parse('3d2h32m', as_timedelta=True), relativedelta(days=3, hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('+3d2h32m', as_timedelta=True), relativedelta(days=3, hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('-3d2h32m', as_timedelta=True), -relativedelta(days=3, hours=2, minutes=32))
+
+    def test_timeparse_4(self):
+        """timeparse test case 4."""
+        self.assertEqual(timeparse.parse('1w3d2h32m', as_timedelta=True), relativedelta(weeks=1, days=3, hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('+1w3d2h32m', as_timedelta=True), relativedelta(weeks=1, days=3, hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('-1w3d2h32m', as_timedelta=True), -relativedelta(weeks=1, days=3, hours=2, minutes=32))
+
+    def test_timeparse_5(self):
+        """timeparse test case 5."""
+        self.assertEqual(timeparse.parse('1w 3d 2h 32m', as_timedelta=True), relativedelta(weeks=1, days=3, hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('+1w 3d 2h 32m', as_timedelta=True), relativedelta(weeks=1, days=3, hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('-1w 3d 2h 32m', as_timedelta=True), -relativedelta(weeks=1, days=3, hours=2, minutes=32))
+
+    def test_timeparse_6(self):
+        """timeparse test case 6."""
+        self.assertEqual(timeparse.parse('1 w 3 d 2 h 32 m', as_timedelta=True), relativedelta(weeks=1, days=3, hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('+1 w 3 d 2 h 32 m', as_timedelta=True), relativedelta(weeks=1, days=3, hours=2, minutes=32))
+        self.assertEqual(timeparse.parse('-1 w 3 d 2 h 32 m', as_timedelta=True), -relativedelta(weeks=1, days=3, hours=2, minutes=32))
+
+    def test_timeparse_7(self):
+        """timeparse test case 7."""
+        self.assertEqual(timeparse.parse('4:13', as_timedelta=True), relativedelta(minutes=4, seconds=13))
+        self.assertEqual(timeparse.parse('+4:13', as_timedelta=True), relativedelta(minutes=4, seconds=13))
+        self.assertEqual(timeparse.parse('-4:13', as_timedelta=True), -relativedelta(minutes=4, seconds=13))
+
+    def test_timeparse_bare_seconds(self):
+        """timeparse test bare seconds, without minutes."""
+        self.assertEqual(timeparse.parse(':13', as_timedelta=True), relativedelta(seconds=13))
+        self.assertEqual(timeparse.parse('+:13', as_timedelta=True), relativedelta(seconds=13))
+        self.assertEqual(timeparse.parse('-:13', as_timedelta=True), -relativedelta(seconds=13))
+
+    def test_timeparse_8(self):
+        """timeparse test case 8."""
+        self.assertEqual(timeparse.parse('4:13:02', as_timedelta=True), relativedelta(hours=4, minutes=13, seconds=2))
+        self.assertEqual(timeparse.parse('+4:13:02', as_timedelta=True), relativedelta(hours=4, minutes=13, seconds=2))
+        self.assertEqual(timeparse.parse('-4:13:02', as_timedelta=True), relativedelta(hours=4, minutes=13, seconds=2))
+
+    def test_timeparse_9(self):
+        """timeparse test case 9."""
+        self.assertEqual(timeparse.parse('4:13:02.266', as_timedelta=True), relativedelta(hours=4, minutes=13, seconds=2, microseconds=266*1000))
+        self.assertEqual(timeparse.parse('+4:13:02.266', as_timedelta=True), relativedelta(hours=4, minutes=13, seconds=2, microseconds=266*1000))
+        self.assertEqual(timeparse.parse('-4:13:02.266', as_timedelta=True), -relativedelta(hours=4, minutes=13, seconds=2, microseconds=266*1000))
+
+    def test_timeparse_10(self):
+        """timeparse test case 10."""
+        self.assertEqual(timeparse.parse('2:04:13:02.266', as_timedelta=True), relativedelta(days=2, hours=4, minutes=13, seconds=2, microseconds=266*1000))
+        self.assertEqual(timeparse.parse('+2:04:13:02.266', as_timedelta=True), relativedelta(days=2, hours=4, minutes=13, seconds=2, microseconds=266*1000))
+        self.assertEqual(timeparse.parse('-2:04:13:02.266', as_timedelta=True), -relativedelta(days=2, hours=4, minutes=13, seconds=2, microseconds=266*1000))
+
+    def test_timeparse_granularity_1(self):
+        """Check that minute-level granularity applies correctly."""
+        self.assertEqual(timeparse.parse('4:32', granularity='minutes', as_timedelta=True), relativedelta(hours=4, minutes=32))
+        self.assertEqual(timeparse.parse('+4:32', granularity='minutes', as_timedelta=True), relativedelta(hours=4, minutes=32))
+        self.assertEqual(timeparse.parse('-4:32', granularity='minutes', as_timedelta=True), -relativedelta(hours=4, minutes=32))
+
+    def test_timeparse_granularity_2(self):
+        """Check that minute-level granularity does not apply inappropriately."""
+        self.assertEqual(timeparse.parse('4:32:02', granularity='minutes', as_timedelta=True), relativedelta(hours=4, minutes=32, seconds=2))
+        self.assertEqual(timeparse.parse('+4:32:02', granularity='minutes', as_timedelta=True), relativedelta(hours=4, minutes=32, seconds=2))
+        self.assertEqual(timeparse.parse('-4:32:02', granularity='minutes', as_timedelta=True), -relativedelta(hours=4, minutes=32, seconds=2))
+
+    def test_timeparse_granularity_3(self):
+        """Check that minute-level granularity does not apply inappropriately."""
+        self.assertEqual(timeparse.parse('7:02.223', granularity='minutes', as_timedelta=True), relativedelta(minutes=7, seconds=2, microseconds=1000*223))
+        self.assertEqual(timeparse.parse('+7:02.223', granularity='minutes', as_timedelta=True), relativedelta(minutes=7, seconds=2, microseconds=1000*223))
+        self.assertEqual(timeparse.parse('-7:02.223', granularity='minutes', as_timedelta=True), -relativedelta(minutes=7, seconds=2, microseconds=1000*223))
+
+    def test_timeparse_granularity_4(self):
+        """Check that minute-level granularity does not apply inappropriately."""
+        self.assertEqual(timeparse.parse('0:02', granularity='seconds', as_timedelta=True), relativedelta(seconds=2))
+        self.assertEqual(timeparse.parse('+0:02', granularity='seconds', as_timedelta=True), relativedelta(seconds=2))
+        self.assertEqual(timeparse.parse('-0:02', granularity='seconds', as_timedelta=True), -relativedelta(seconds=2))
+
+    def test_timeparse_unparsed(self):
+        """Check that unparsed values tries to converts into int(). """
+        self.assertEqual(timeparse.parse(100, as_timedelta=True), relativedelta(seconds=100))
+        self.assertEqual(timeparse.parse(-18.333, as_timedelta=True), -relativedelta(seconds=18, microseconds=333*1000))
+        self.assertEqual(timeparse.parse('99.1', as_timedelta=True), relativedelta(seconds=99, microseconds=100*1000))
+        self.assertEqual(timeparse.parse('-99.1', as_timedelta=True), -relativedelta(seconds=99, microseconds=100*1000))
+
+    def test_timeparse_11(self):
+        """timeparse test case 11."""
+        # uptime format
+        self.assertEqual(timeparse.parse('2 days,  4:13:02', as_timedelta=True), relativedelta(days=2, hours=4, minutes=13, seconds=2))
+        self.assertEqual(timeparse.parse('+2 days,  4:13:02', as_timedelta=True), relativedelta(days=2, hours=4, minutes=13, seconds=2))
+        self.assertEqual(timeparse.parse('-2 days,  4:13:02', as_timedelta=True), -relativedelta(days=2, hours=4, minutes=13, seconds=2))
+
+    def test_timeparse_12(self):
+        """timeparse test case 12."""
+        self.assertEqual(timeparse.parse('2 days,  4:13:02.266', as_timedelta=True), relativedelta(days=2, hours=4, minutes=13, seconds=2, microseconds=1000*266))
+        self.assertEqual(timeparse.parse('+2 days,  4:13:02.266', as_timedelta=True), relativedelta(days=2, hours=4, minutes=13, seconds=2, microseconds=1000*266))
+        self.assertEqual(timeparse.parse('-2 days,  4:13:02.266', as_timedelta=True), -relativedelta(days=2, hours=4, minutes=13, seconds=2, microseconds=1000*266))
+
+    def test_timeparse_13(self):
+        """timeparse test case 13."""
+        self.assertEqual(timeparse.parse('5hr34m56s', as_timedelta=True), relativedelta(hours=5, minutes=34, seconds=56))
+        self.assertEqual(timeparse.parse('+5hr34m56s', as_timedelta=True), relativedelta(hours=5, minutes=34, seconds=56))
+        self.assertEqual(timeparse.parse('-5hr34m56s', as_timedelta=True), -relativedelta(hours=5, minutes=34, seconds=56))
+
+    def test_timeparse_14(self):
+        """timeparse test case 14."""
+        self.assertEqual(timeparse.parse('5 hours, 34 minutes, 56 seconds', as_timedelta=True), relativedelta(hours=5, minutes=34, seconds=56))
+        self.assertEqual(timeparse.parse('+5 hours, 34 minutes, 56 seconds', as_timedelta=True), relativedelta(hours=5, minutes=34, seconds=56))
+        self.assertEqual(timeparse.parse('-5 hours, 34 minutes, 56 seconds', as_timedelta=True), -relativedelta(hours=5, minutes=34, seconds=56))
+
+    def test_timeparse_15(self):
+        """timeparse test case 15."""
+        self.assertEqual(timeparse.parse('5 hrs, 34 mins, 56 secs', as_timedelta=True), relativedelta(hours=5, minutes=34, seconds=56))
+        self.assertEqual(timeparse.parse('+5 hrs, 34 mins, 56 secs', as_timedelta=True), relativedelta(hours=5, minutes=34, seconds=56))
+        self.assertEqual(timeparse.parse('-5 hrs, 34 mins, 56 secs', as_timedelta=True), -relativedelta(hours=5, minutes=34, seconds=56))
+
+    def test_timeparse_16(self):
+        """timeparse test case 16."""
+        self.assertEqual(timeparse.parse('2 days, 5 hours, 34 minutes, 56 seconds', as_timedelta=True), relativedelta(days=2, hours=5, minutes=34, seconds=56))
+        self.assertEqual(timeparse.parse('+2 days, 5 hours, 34 minutes, 56 seconds', as_timedelta=True), relativedelta(days=2, hours=5, minutes=34, seconds=56))
+        self.assertEqual(timeparse.parse('-2 days, 5 hours, 34 minutes, 56 seconds', as_timedelta=True), -relativedelta(days=2, hours=5, minutes=34, seconds=56))
+
+    def test_timeparse_16b(self):
+        """timeparse test case 16b."""
+        self.assertEqual(timeparse.parse('1.75 s', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('+1.75 s', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('-1.75 s', as_timedelta=True), -relativedelta(seconds=1, microseconds=75*10000))
+
+    def test_timeparse_16c(self):
+        """timeparse test case 16c."""
+        self.assertEqual(timeparse.parse('1.75 sec', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('+1.75 sec', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('-1.75 sec', as_timedelta=True), -relativedelta(seconds=1, microseconds=75*10000))
+
+    def test_timeparse_16d(self):
+        """timeparse test case 16d."""
+        self.assertEqual(timeparse.parse('1.75 secs', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('+1.75 secs', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('-1.75 secs', as_timedelta=True), -relativedelta(seconds=1, microseconds=75*10000))
+
+    def test_timeparse_16e(self):
+        """timeparse test case 16e."""
+        self.assertEqual(timeparse.parse('1.75 second', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('+1.75 second', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('-1.75 second', as_timedelta=True), -relativedelta(seconds=1, microseconds=75*10000))
+
+    def test_timeparse_16f(self):
+        """timeparse test case 16f."""
+        self.assertEqual(timeparse.parse('1.75 seconds', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('+1.75 seconds', as_timedelta=True), relativedelta(seconds=1, microseconds=75*10000))
+        self.assertEqual(timeparse.parse('-1.75 seconds', as_timedelta=True), -relativedelta(seconds=1, microseconds=75*10000))
+
+    def test_timeparse_17(self):
+        """timeparse test case 17."""
+        self.assertEqual(timeparse.parse('1.2 m', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('+1.2 m', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('-1.2 m', as_timedelta=True), -relativedelta(minutes=1, seconds=12))
+
+    def test_timeparse_18(self):
+        """timeparse test case 18."""
+        self.assertEqual(timeparse.parse('1.2 min', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('+1.2 min', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('-1.2 min', as_timedelta=True), -relativedelta(minutes=1, seconds=12))
+
+    def test_timeparse_19(self):
+        """timeparse test case 19."""
+        self.assertEqual(timeparse.parse('1.2 mins', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('+1.2 mins', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('-1.2 mins', as_timedelta=True), -relativedelta(minutes=1, seconds=12))
+
+    def test_timeparse_20(self):
+        """timeparse test case 20."""
+        self.assertEqual(timeparse.parse('1.2 minute', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('+1.2 minute', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('-1.2 minute', as_timedelta=True), -relativedelta(minutes=1, seconds=12))
+
+    def test_timeparse_21(self):
+        """timeparse test case 21."""
+        self.assertEqual(timeparse.parse('1.2 minutes', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('+1.2 minutes', as_timedelta=True), relativedelta(minutes=1, seconds=12))
+        self.assertEqual(timeparse.parse('-1.2 minutes', as_timedelta=True), -relativedelta(minutes=1, seconds=12))
+
+    def test_timeparse_22(self):
+        """timeparse test case 22."""
+        self.assertEqual(timeparse.parse('172 hours', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('+172 hours', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('-172 hours', as_timedelta=True), -relativedelta(hours=172))
+
+    def test_timeparse_23(self):
+        """timeparse test case 23."""
+        self.assertEqual(timeparse.parse('172 hr', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('+172 hr', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('-172 hr', as_timedelta=True), -relativedelta(hours=172))
+
+    def test_timeparse_24(self):
+        """timeparse test case 24."""
+        self.assertEqual(timeparse.parse('172 h', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('+172 h', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('-172 h', as_timedelta=True), -relativedelta(hours=172))
+
+    def test_timeparse_25(self):
+        """timeparse test case 25."""
+        self.assertEqual(timeparse.parse('172 hrs', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('+172 hrs', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('-172 hrs', as_timedelta=True), -relativedelta(hours=172))
+
+    def test_timeparse_26(self):
+        """timeparse test case 26."""
+        self.assertEqual(timeparse.parse('172 hour', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('+172 hour', as_timedelta=True), relativedelta(hours=172))
+        self.assertEqual(timeparse.parse('-172 hour', as_timedelta=True), -relativedelta(hours=172))
+
+    def test_timeparse_27(self):
+        """timeparse test case 27."""
+        self.assertEqual(timeparse.parse('1.24 days', as_timedelta=True), relativedelta(days=1, hours=5, minutes=45, seconds=36))
+        self.assertEqual(timeparse.parse('+1.24 days', as_timedelta=True), relativedelta(days=1, hours=5, minutes=45, seconds=36))
+        self.assertEqual(timeparse.parse('-1.24 days', as_timedelta=True), -relativedelta(days=1, hours=5, minutes=45, seconds=36))
+
+    def test_timeparse_28(self):
+        """timeparse test case 28."""
+        self.assertEqual(timeparse.parse('5 d', as_timedelta=True), relativedelta(days=5))
+        self.assertEqual(timeparse.parse('+5 d', as_timedelta=True), relativedelta(days=5))
+        self.assertEqual(timeparse.parse('-5 d', as_timedelta=True), -relativedelta(days=5))
+
+    def test_timeparse_29(self):
+        """timeparse test case 29."""
+        self.assertEqual(timeparse.parse('5 day', as_timedelta=True), relativedelta(days=5))
+        self.assertEqual(timeparse.parse('+5 day', as_timedelta=True), relativedelta(days=5))
+        self.assertEqual(timeparse.parse('-5 day', as_timedelta=True), -relativedelta(days=5))
+
+    def test_timeparse_30(self):
+        """timeparse test case 30."""
+        self.assertEqual(timeparse.parse('5 days', as_timedelta=True), relativedelta(days=5))
+        self.assertEqual(timeparse.parse('+5 days', as_timedelta=True), relativedelta(days=5))
+        self.assertEqual(timeparse.parse('-5 days', as_timedelta=True), -relativedelta(days=5))
+
+    def test_timeparse_31(self):
+        """timeparse test case 31."""
+        self.assertEqual(timeparse.parse('5.6 wk', as_timedelta=True), relativedelta(days=39, hours=4, minutes=48))
+        self.assertEqual(timeparse.parse('+5.6 wk', as_timedelta=True), relativedelta(days=39, hours=4, minutes=48))
+        self.assertEqual(timeparse.parse('-5.6 wk', as_timedelta=True), -relativedelta(days=39, hours=4, minutes=48))
+
+    def test_timeparse_32(self):
+        """timeparse test case 32."""
+        self.assertEqual(timeparse.parse('5.6 week', as_timedelta=True), relativedelta(days=39, hours=4, minutes=48))
+        self.assertEqual(timeparse.parse('+5.6 week', as_timedelta=True), relativedelta(days=39, hours=4, minutes=48))
+        self.assertEqual(timeparse.parse('-5.6 week', as_timedelta=True), -relativedelta(days=39, hours=4, minutes=48))
+
+    def test_timeparse_33(self):
+        """timeparse test case 33."""
+        self.assertEqual(timeparse.parse('5.6 weeks', as_timedelta=True), relativedelta(days=39, hours=4, minutes=48))
+        self.assertEqual(timeparse.parse('+5.6 weeks', as_timedelta=True), relativedelta(days=39, hours=4, minutes=48))
+        self.assertEqual(timeparse.parse('-5.6 weeks', as_timedelta=True), -relativedelta(days=39, hours=4, minutes=48))
+
+    def test_milliseconds(self):
+        self.assertEqual(timeparse.parse('3 ms', as_timedelta=True), relativedelta(microseconds=3000))
+        self.assertEqual(timeparse.parse('3 millis', as_timedelta=True), relativedelta(microseconds=3000))
+        self.assertEqual(timeparse.parse('3 msecs', as_timedelta=True), relativedelta(microseconds=3000))
+        self.assertEqual(timeparse.parse('3 milliseconds', as_timedelta=True), relativedelta(microseconds=3000))
+
+    def test_plain_numbers(self):
+        self.assertEqual(timeparse.parse('10', as_timedelta=True), relativedelta(seconds=10))
+        self.assertEqual(timeparse.parse('10.1', as_timedelta=True), relativedelta(seconds=10, microseconds=100000))
+        self.assertEqual(timeparse.parse('-10', as_timedelta=True), -relativedelta(seconds=10))
+        self.assertEqual(timeparse.parse('-10.1', as_timedelta=True), -relativedelta(seconds=10, microseconds=100000))
+
+    def test_combined(self):
+        self.assertEqual(timeparse.parse('1y2mo3w4d5h6m7s8ms', as_timedelta=True), relativedelta(years=1, months=2, weeks=3, days=4, hours=5, minutes=6, seconds=7, microseconds=8000))
+
+
+
+class MiscTests(unittest.TestCase):
+    """
+    Miscellaneous unit tests for the `timeparse` module.
+    """
+
     def test_strange(self):
         self.assertIsNone(timeparse.parse('1.1.1:22'))
 
@@ -441,6 +758,8 @@ class TestTimeparse(unittest.TestCase):
         self.assertIsInstance(timeparse.parse('10:10', as_timedelta=True), datetime.timedelta)
         timeparse.enable_dateutil()
         self.assertNotIsInstance(timeparse.parse('10:10', as_timedelta=True), datetime.timedelta)
+
+
 
 
 if __name__ == '__main__':

